tree/treeplayer/src/TTreeReaderValue.cxx

```diff
@@ -601,52 +602,35 @@ void ROOT::Internal::TTreeReaderValueBase::CreateProxy() {
             membername = branch->GetName();
          }
       }
-      auto director = fTreeReader->fDirector;
+      auto director = fTreeReader->fDirector.get();
       // Determine if the branch is actually in a Friend TTree and if so which.
       if (branch->GetTree() != fTreeReader->GetTree()->GetTree()) {
          // It is in a friend, let's find the 'index' in the list of friend ...
          int index = -1;
-         int current = 0;
-         TFriendElement *fe_found = nullptr;
-         for(auto fe : TRangeDynCast<TFriendElement>( fTreeReader->GetTree()->GetTree()->GetListOfFriends())) {
-            if (branch->GetTree() == fe->GetTree()) {
-               index = current;
-               fe_found = fe;
-               break;
-            }
-            ++current;
+         const auto friendElements =
+            TRangeDynCast<TFriendElement>(fTreeReader->GetTree()->GetTree()->GetListOfFriends());
+         if (auto foundFriend =
+                std::find_if(friendElements.begin(), friendElements.end(),
+                             [&branch](TFriendElement *fe) { return branch->GetTree() == fe->GetTree(); });
+             foundFriend != friendElements.end()) {
+            index = static_cast<int>(std::distance(friendElements.begin(), foundFriend));
          }
          if (index == -1) {
-            Error(errPrefix, "The branch %s is contained in a Friend TTree that is not directly attached to the main.\n"
+            Error(errPrefix,
+                  "The branch %s is contained in a Friend TTree that is not directly attached to the main.\n"
                   "This is not yet supported by TTreeReader.",
                   fBranchName.Data());
             return;
          }
-         const char *localBranchName =  originalBranchName.c_str();
```

Comment:
Why is this not used?

Reply:
I don't know. It was introduced in [this commit](https://github.com/root-project/root/commit/d35d36eff99d3e9d4881c19e86a98813b743e0c9) and only for TTreeReaderValue, TTreeReaderArray doesn't have that bit. The variable is never referenced anywhere in the function body, nor is this logic used anywhere else in TTreeReader code.

Reply:
From reading that code, it is used in the case there user provided a long name (`friendtreename.branchname`) to access a branch and we want to extract/find the branchname ...

However indeed, the result is not used ...